cmake/module/GetTargetInterface.cmake

```diff
@@ -32,7 +32,18 @@ endfunction()
 function(get_target_interface var config target property)
   get_target_property(result ${target} INTERFACE_${property})
   if(result)
-    evaluate_generator_expressions(result "${config}")
+    # On systems where pthread functionality is not provided by
+    # the C library implementation, the CMake FindThreads module
+    # sets the Threads::Threads target's compile options to
+    # generator expressions that evaluate to `-pthread` in this
+    # project.
```

Comment:
> generator expressions that evaluate to `-pthread` in this project.

So couldn't we just stop using `Threads::Threads` and use `-pthread` directly?

Reply:
1. It is our practise to check whether flags are supported and required, isn't it?
2. In the Autotools-based build system, we used the [`AX_PTHREAD`](https://www.gnu.org/software/autoconf-archive/ax_pthread.html) macro for that purpose.
3. The [`FindThreads`](https://cmake.org/cmake/help/latest/module/FindThreads.html) module is a CMake's way of achieving the same.
4. The current issue on the master branch is not about the thread-related flags for the toolchain but rather their presentation in the summary.